tests/test_game.py

```diff
@@ -63,53 +67,60 @@ def test_get_board_state_sends_a_message_to_board():
     board = MockBoard()
     game = Game(board)
     game.get_board_state()
-    assert board.get_spaces_call_count == 1
 
-
-def test_returns_current_player(game):
-    current_player_at_start = game.get_player_one()
-    assert game.get_current_player() == current_player_at_start
+    assert board.get_spaces_call_count == 1
 
 
 def test_current_player_switches_to_player_two_after_player_one_turn(game):
     current_player_after_switch = game.get_player_two()
     game.play_turn(0)
+
     assert game.get_current_player() == current_player_after_switch
 
 
 def test_current_player_reverts_to_player_one_after_player_two_turn(game):
     current_player_on_turn_three = game.get_player_one()
     game.play_turn(0)
     game.play_turn(1)
+
     assert game.get_current_player() == current_player_on_turn_three
 
 
-def test_the_game_is_not_over_when_board_is_not_full(game):
+def test_game_is_not_over_when_board_is_not_full(game):
     assert game.game_over() is False
 
 
 def test_the_game_is_over_when_board_is_full():
-    end_state_game = Game(game_board=MockEndStateBoard())
+    end_state_game = Game(game_board=MockFullBoard())
+
     assert end_state_game.game_over() is True
 
 
+def test_the_game_is_over_if_a_player_has_won():
+    won_game = Game(game_board=MockBoardWithWinConditionMet())
+
+    assert won_game.game_over() is True
+
+
 def test_play_turn_instructs_board_to_occupy_space():
     board = MockBoard()
     game = Game(board)
     game.play_turn('1')
+
     assert board.occupy_space_call_count == 1
 
 
 def test_play_turn_raises_an_error_for_if_passed_an_occupied_space(game):
-    occupied_space = '8'
+    occupied_space = '3'
```

Comment:
Ignoring the way that pytest forces us to catch the exception, does this follow the 4 phase test (Arrange-Act-Assert-Tear Down)?

What do you think about the following test:
```
game.play_turn(3)
with pytest.raises(Exception) as err:
    game.play_turn(3)
assert "Invalid Move!" in str(err.value)
```

What are the pros and cons of each approach?